server/spinwick.go

```diff
@@ -89,46 +93,110 @@ func (s *Server) createInstallationRequest(ownerID, version, image, dns, size st
+	sysadminPassword, userPassword, err := s.initializeMattermostTestServer(spinwickURL, pr.Number, logger)
 	if err != nil {
-		return errors.Wrap(err, "failed to initialize the Installation")
+		return "", "", errors.Wrap(err, "failed to initialize the Installation")
 	}
 
-	return nil
+	return sysadminPassword, userPassword, nil
 }
 
-// Helper function to format and send success comment
-func (s *Server) sendSpinwickSuccessComment(pr *model.PullRequest, installation *cloudModel.InstallationDTO, extraInfo string) {
+// Helper function to generate a secure random password
+func generateSecurePassword() (string, error) {
+	bytes := make([]byte, 16)
+	if _, err := rand.Read(bytes); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes), nil
+}
+
+// Helper function to format and send success comment to Mattermost webhook
+func (s *Server) sendSpinwickSuccessToMattermost(pr *model.PullRequest, installation *cloudModel.InstallationDTO, sysadminPassword, userPassword, extraInfo string, logger logrus.FieldLogger) {
+	// Send public message to GitHub (without credentials)
 	spinwickURL := fmt.Sprintf("https://%s", cloudtools.GetInstallationDNSFromDNSRecords(installation))
-	userTable := "| Account Type | Username | Password |\n|---|---|---|\n| Admin | sysadmin | Sys@dmin123 |\n| User | user-1 | User-1@123 |"
 	logLink := fmt.Sprintf("https://grafana.internal.mattermost.com/explore?orgId=1&left=%%7B%%22datasource%%22:%%22PFB2D5CACEC34D62E%%22,%%22queries%%22:%%5B%%7B%%22refId%%22:%%22A%%22,%%22expr%%22:%%22%%7Bnamespace%%3D%%5C%%22%s%%5C%%22%%7D%%22,%%22queryType%%22:%%22range%%22,%%22datasource%%22:%%7B%%22type%%22:%%22loki%%22,%%22uid%%22:%%22PFB2D5CACEC34D62E%%22%%7D,%%22editorMode%%22:%%22code%%22%%7D%%5D,%%22range%%22:%%7B%%22from%%22:%%22now-1h%%22,%%22to%%22:%%22now%%22%%7D%%7D", installation.ID)
 
-	baseMsg := fmt.Sprintf("test server created! :tada:\n\nAccess here: %s\n\n%s", spinwickURL, userTable)
-	if extraInfo != "" {
-		baseMsg += "\n\n" + extraInfo
-	}
-	baseMsg += fmt.Sprintf("\n\nYour Spinwick's installation ID is: `%s`\nTo access the logs, please click [here](%s)", installation.ID, logLink)
-
 	var prefix string
 	if s.isPluginRepository(pr.RepoName) {
 		prefix = "Plugin "
 	} else {
 		prefix = "Mattermost "
 	}
 
-	s.sendGitHubComment(pr.RepoOwner, pr.RepoName, pr.Number, prefix+baseMsg)
+	// Public GitHub message (no credentials)
+	githubMsg := fmt.Sprintf("%sSpinwick PR #%d :tada:\n\n**Test server created!**\n\nAccess here: %s", prefix, pr.Number, spinwickURL)
+	if extraInfo != "" {
+		githubMsg += "\n\n" + extraInfo
+	}
+	githubMsg += fmt.Sprintf("\n\n**Installation ID:** `%s`\n**Logs:** [Click here](%s)", installation.ID, logLink)
+
+	// Add link to credentials channel if configured
+	if s.Config.MattermostCredentialsChannelURL != "" {
+		githubMsg += fmt.Sprintf("\n\n**Credentials:** Posted securely in [this Mattermost channel](%s) - Look for PR #%d", s.Config.MattermostCredentialsChannelURL, pr.Number)
```

Comment:
Do we get the `post_id` back from the webhook? Might we permalink directly to the post in question?

Reply:
I was surprised at this, but no, we don't. It just returns a plain text `ok` response. We'd have to create through the proper post API to get it, maybe a future improvement.